pandas/core/arrays/masked.py

```diff
@@ -108,6 +108,13 @@ class BaseMaskedArray(OpsMixin, ExtensionArray):
     _truthy_value = Scalar  # bool(_truthy_value) = True
     _falsey_value = Scalar  # bool(_falsey_value) = False
 
+    @classmethod
+    def _simple_new(cls, values: np.ndarray, mask: npt.NDArray[np.bool_]) -> Self:
+        result = BaseMaskedArray.__new__(cls)
+        result._data = values
+        result._mask = mask
+        return result
```

Comment:
This is from #53013 (I built this on top of that to see what performance improvements I get), so can be handled independently in #53013.

and yes, this is to avoid the checks, when not needed. For this PR i need `BaseMaskedArray.reshape` to be fast and this helps with that (and many other cases).